### DISCUSSION TITLE
Cannot expand home directory to choose folder for either media or playlists.ary

### DISCUSSION THREAD

**Discussion by w06lmy** [LMS-Community/slimserver#870] [General]
During initial setup and and also when I visit settings after completing the initial setup, I am unable to expand the home folder in order select subfolders in within the /home directory.   I am able to expand other folders that contain subfolders.  I am running linux, so in the home directory are subfolders for users and, for each user, at least a Music folder.  I want to use /home/username/Music, but cannot access it in the server settings.  If I click the plus sign to expand, it disappears, and home is selected.  Not acceptable.  If I click on other folders in the root directory that contain subfolders, I am able to open all the way down to the lowest level that doesn't contain a subfolder.

I am running kubuntu 2022.04.1.  I had problems installing it, as discussed in another thread, so I assumed a faulty installation.  I installed Fedora 37, which installed with no problem, but I also cannot access subfolders of the home directory with Fedora.

I am trying to install in my own /home/user/Music directory.

**Comment by michaelherger** [comment]
Most likely you're not looking at your home, but the `slimserver` user's - which might be empty. LMS is not running as your user.

**Comment by w06lmy** [comment]
/home is the folder containing all users' home pages.  No subfolders, such as /home/user1 or /home/user2 are visible when I click on the plus sign for the listing "home" that appears in the dialog box for selecting a media folder.

I have confirmed that the user home directory /home/user/Music does appear.  I confirmed that on my previous installation of LMS, on kubuntu 2018.04, LMS is using the music folder in my home directory, /home/user/Music.   

I am attaching two images.  Screenshot 1 shows the dialog box for selecting a folder before I select home, and screenshot 2 shows the dialog box after I select home.  

![Screenshot_of_folder_choicesf_2](https://user-images.githubusercontent.com/65308112/224450863-078fbeb0-1a0c-4306-b47a-39620e9c8c74.png)

![Screenshot_of_folder_choicesf_1](https://user-images.githubusercontent.com/65308112/224450834-ecd4b7ca-5746-40e9-9f0f-047a4da72a4a.png)

**Comment by michaelherger** [comment]
It's a permissions issue. You have to grant the LMS user full access to your home folder, or it won't be able to access it.

**Reply by w06lmy** [reply]
I am a mere mortal linux user, understanding just enough about permissions to be dangerous.

I have just installed LMS 8.3.1 on an identical computer running kubuntu 2018 without encountering this problem.  Why now, with kubuntu 2022?  Has ubuntu added some additional security?  If so, then Fedora must have done it too, because I had the same problem installing the LMS rpm to fedora.

As to what to do about it.  I don't see an account on the machine for LMS, so how can I give it permission to my music folder?  I found a suggestion on line that I give permission to the folder to all users, and I suppose that would be better than to use the no-longer-supported version of kubuntu on which LMS is running and working, but there must be a better way.  That approach also has the disadvantage that I will have to change permissions to new files when I add them.

**Comment by w06lmy** [comment]
Apparently kubuntu changed its default permissions for home directories:
ubuntu 2018:   drwsr-xr-x
ubuntu 2022:   drwsr-x--- 

Interestingly, all of the default folders in the home directory had the permissions drwxr-xr-x.   

Problem solved by applying the 2018 permissions to the 2022 installation.  I see that Fedora puts even more severe restrictions on the home directory, drwx------.

Would it be possible to put LMS in a group and give it permission that way?  How would you assign LMS to a group if it doesn't have an account on the machined?  That's probably a linux question and off topic.
